From fbf6df2696fd540bc19303c1655d9dabef769e7f Mon Sep 17 00:00:00 2001
From: Jacky Bai <ping.bai@nxp.com>
Date: Mon, 3 Aug 2020 13:31:26 +0800
Subject: [PATCH] MLK-24457 plat: imx8m: Fix the current fsp init

The dfimisc reg value should be shift right 8 bit to
get the current fsp.

Signed-off-by: Jacky Bai <ping.bai@nxp.com>
Reviewed-by: Anson Huang <Anson.Huang@nxp.com>
---
 plat/imx/imx8m/ddr/dram.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/plat/imx/imx8m/ddr/dram.c b/plat/imx/imx8m/ddr/dram.c
index b206c716a..3424e92cd 100644
--- a/plat/imx/imx8m/ddr/dram.c
+++ b/plat/imx/imx8m/ddr/dram.c
@@ -179,7 +179,8 @@ void dram_info_init(unsigned long dram_timing_base)
 		DDRC_ACTIVE_TWO_RANK : DDRC_ACTIVE_ONE_RANK;
 
 	/* Get current fsp info */
-	current_fsp = mmio_read_32(DDRC_DFIMISC(0)) & 0xf;
+	current_fsp = mmio_read_32(DDRC_DFIMISC(0));
+	current_fsp = (current_fsp >> 8) & 0xf;
 	dram_info.boot_fsp = current_fsp;
 	dram_info.current_fsp = current_fsp;
 
@@ -226,7 +227,6 @@ void dram_info_init(unsigned long dram_timing_base)
 	}
 }
 
-
 /*
  * For each freq return the following info:
  *
-- 
2.17.1

